### PR TITLE
[Merged by Bors] - Unwrap removal

### PR DESCRIPTION
**Pull request by Razican**
This removes all the calls to  `unwrap()` in the codebase, which made me found a couple of places where it wasn't needed, and could be improved. I also noticed we don't have dependabot updates for the test262 submodule and the interner dependencies, so I added those.

I added lints so that no new unwraps are added.

### CODE REVIEW

File: boa/src/syntax/parser/expression/mod.rs [hunk 1]
```diff
@@ -489,9 +489,9 @@ where
     fn parse(self, cursor: &mut Cursor<R>, interner: &mut Interner) -> ParseResult {
         let _timer = BoaProfiler::global().start_event("Relation Expression", "Parsing");
 
-        if None::<InputElement>.is_some() {
-            cursor.set_goal(None::<InputElement>.unwrap());
-        }
+        // if None::<InputElement>.is_some() {
+        //     cursor.set_goal(None::<InputElement>.unwrap());
+        // }
```
Comment (RageKnify) [inline boa/src/syntax/parser/expression/mod.rs:494]: What was the point of this?

**Review by RageKnify — APPROVED** [review]

File: boa/src/syntax/parser/expression/mod.rs (hunk 1 above)
Comment (Razican) [reply boa/src/syntax/parser/expression/mod.rs:494]: Oh, leftover code, I was just surprised about the `None.is_some()` check, I will remove it

Comment (RageKnify) [reply boa/src/syntax/parser/expression/mod.rs:494]: But shouldn't it panic unconditionally? `None` is obviously not `Some`

Comment (Razican) [reply boa/src/syntax/parser/expression/mod.rs:494]: It panics always, but it checks before if it's some, so it never enters the branch.

**Pull request closed without merging**